markdown/extension/strikethrough/src/main/kotlin/org/jetbrains/jewel/markdown/extension/strikethrough/StrikethroughRendererExtension.kt

```diff
@@ -0,0 +1,49 @@
+package org.jetbrains.jewel.markdown.extension.strikethrough
+
+import androidx.compose.ui.text.AnnotatedString.Builder
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.style.TextDecoration
+import org.jetbrains.jewel.markdown.InlineMarkdown
+import org.jetbrains.jewel.markdown.WithInlineMarkdown
+import org.jetbrains.jewel.markdown.extensions.MarkdownInlineRendererExtension
+import org.jetbrains.jewel.markdown.extensions.MarkdownRendererExtension
+import org.jetbrains.jewel.markdown.rendering.InlineMarkdownRenderer
+import org.jetbrains.jewel.markdown.rendering.InlinesStyling
+
+public class StrikethroughRendererExtension(private val inlinesStyling: InlinesStyling) : MarkdownRendererExtension {
+    override val inlineRenderer: MarkdownInlineRendererExtension
+        get() =
+            object : MarkdownInlineRendererExtension {
+                override fun canRender(inline: InlineMarkdown.CustomNode): Boolean = inline is CustomStrikethroughNode
+
+                override fun render(
+                    builder: Builder,
+                    inline: InlineMarkdown.CustomNode,
+                    inlineRenderer: InlineMarkdownRenderer,
+                    enabled: Boolean,
+                ) {
+                    with(builder) {
+                        val popTo =
+                            pushStyle(
+                                inlinesStyling.textStyle
+                                    .copy(textDecoration = TextDecoration.LineThrough)
+                                    .toSpanStyle()
+                                    .copy()
+                            )
+                        val inlineMarkdowns = (inline as? WithInlineMarkdown)?.inlineContent ?: emptyList()
+                        for (markdown in inlineMarkdowns) {
+                            append(
+                                inlineRenderer.renderAsAnnotatedString(listOf(markdown), inlinesStyling, enabled = true)
+                            )
+                        }
+                        pop(popTo)
+                    }
```

Comment:
just fyi, you can use `builder.withStyle` to automatically get a push/pop of that style

```suggestion
                   val style =  inlinesStyling.textStyle
                                    .copy(textDecoration = TextDecoration.LineThrough)
                                    .toSpanStyle()
                                    .copy()
                                    
                    builder.withStyle(style) {
                                  val inlineMarkdowns = (inline as? WithInlineMarkdown)?.inlineContent ?: emptyList()
                                  for (markdown in inlineMarkdowns) {
                                      append(
                                          inlineRenderer.renderAsAnnotatedString(listOf(markdown), inlinesStyling, enabled = true)
                                      )
                                  }
                        }
                    }
```